### ISSUE TITLE
Tier settings not working

### ISSUE THREAD

**Issue by gaam24** [fabriziosalmi/proxmox-lxc-autoscale#13]
**Describe the bug**
Even though I use sample tier settings, all lxc containers are scaled to default settings. When debugging the settings for each container it came out that they were default (only added print in `scaling_manager.py` after `config = DEFAULTS.get(f"TIER_{ctid}", DEFAULTS)`).

**To Reproduce**
1. Add Tier in '/etc/lxc_autoscale/lxc_autoscale.yaml'
2. Reload service

**Additional context**
Python 3.11.2
Kernel Version: Linux 6.8.12-4-pve (2024-11-06T15:04Z)
Manager Version: pve-manager/8.2.8/a577cfa684c7476d

[logs.txt](https://github.com/user-attachments/files/17787416/logs.txt)
[lxc_autoscale.txt](https://github.com/user-attachments/files/17787419/lxc_autoscale.txt)

**Comment by fabriziosalmi** [comment]
THank you to point me out to that, I will try to fix latest 2 issues togheter in my next free time slot, if someone wanna contribute in the meanwhile it will be appreciated!

**Comment by fabriziosalmi** [comment]
Hopefully fixed with [this commit](https://github.com/fabriziosalmi/proxmox-lxc-autoscale/commit/d9a9a26b5f7fd6d17a20ba951de7932dd060362d)

**Comment by pear235** [comment]
I don't think it's working, I set up a tier for a LXC to have at least 1024MB memory, but upon restarting the script it decreases to minimum set in the Default section - 512MB

The Tier section is after the Default one in the yaml config file

**Comment by fabriziosalmi** [comment]
Thank you to note this, I will keep u posted here!

**Comment by pear235** [comment]
Thanks! I am also having issues with two other things:

- CPU scale down to less cores than the "min_cores" value is set - this i was able to fix by modifying the following:
            if new_cores >= min_cores: 
into
            if new_cores > min_cores: 
in the scaling_manager.py

- RAM size doesn't decrease, even if usage is at less that 1% - i tried playing with min_decrease_chunk and / or memory thresholds, but no luck unfortunately

**Comment by fabriziosalmi** [comment]
warm welcome :))

**Comment by gaam24** [comment]
I agree with pear, I tested this yesterday and have this same issues.

**Comment by fabriziosalmi** [comment]
Should be fixed by using this approach(https://github.com/fabriziosalmi/proxmox-lxc-autoscale/commit/a62c82a4ec3e3a9b85d960509bef785b8e8538ce).

On my Proxmox node is working as expected also respect minimum ram settings.

**Comment by pear235** [comment]
I don't think this fixed it :|

Example:

[lxc_autoscale.json](https://github.com/user-attachments/files/18463404/lxc_autoscale.json)
[lxc_autoscale.log](https://github.com/user-attachments/files/18463403/lxc_autoscale.log)
[lxc_autoscale.yaml.log](https://github.com/user-attachments/files/18463405/lxc_autoscale.yaml.log)

TIER_plex:
  lxc_containers:
    - 101

  cpu_upper_threshold: 80
  cpu_lower_threshold: 30
  memory_upper_threshold: 80
  memory_lower_threshold: 30
  min_cores: 2
  max_cores: 6
  min_memory: 1024
  core_min_increment: 1
  core_max_increment: 1
  memory_min_increment: 128
  min_decrease_chunk: 128

- After few executions - Started at 2GB RAM and 2 CPUs:

2025-01-18 07:51:32 - INFO - Container 101: CPU usage: 0.0%, Memory usage: 30.12MB (97.65% free of 1280MB total), Initial cores: 1, Initial memory: 1280MB

So the memory decrease appear to be working, however it ignores min_cores for the tier

===========================================================================



TIER_emby:
  lxc_containers:
    - 102
  cpu_upper_threshold: 80
  cpu_lower_threshold: 30
  memory_upper_threshold: 80
  memory_lower_threshold: 30
  min_cores: 1
  max_cores: 6
  min_memory: 1024
  core_min_increment: 1
  core_max_increment: 1
  memory_min_increment: 128
  min_decrease_chunk: 128

- After few executions - Started at 8GB RAM and 6 CPUs:

2025-01-18 07:51:32 - INFO - Container 102: CPU usage: 68.5%, Memory usage: 22.98MB (99.67% free of 6912MB total), Initial cores: 6, Initial memory: 6912MB

========================================================================



TIER_jellyfin:
  lxc_containers:
    - 115
  cpu_upper_threshold: 80
  cpu_lower_threshold: 40
  memory_upper_threshold: 80
  memory_lower_threshold: 30
  min_cores: 1
  max_cores: 6
  min_memory: 1024
  core_min_increment: 1
  core_max_increment: 1
  memory_min_increment: 128
  min_decrease_chunk: 128

- After few executions - Started at 4GB RAM and 6 CPUs:

2025-01-18 08:13:59 - INFO - Container 115: CPU usage: 40.0%, Memory usage: 52.88MB (98.71% free of 4096MB total), Initial cores: 5, Initial memory: 4096MB

So here it decreased CPU cores count, but didn't even touch the memory.




BTW, are the CPU stats in PVE console the same as those the scripts is seeing? 

In shell i see that CPU usage for CT115 is less than 2% : 
lxc/115                           │ lxc     │ 1.26% │   28.61 GiB │ 122.69 GiB │ 278.57 GiB │         │       │      │      5

Whereas in the scripts log file:

2025-01-18 08:23:00 - INFO - Container 115: CPU usage: 40.0%, Memory usage: 46.91MB (98.85% free of 4096MB total), Initial cores: 5, Initial memory: 4096MB

**Comment by fabriziosalmi** [comment]
Finally I found some time to face that issue, I will push fixes this weekend ☕️

**Comment by fabriziosalmi** [comment]
🍻 since it's saturday 

fixed, please check [configuration update](https://github.com/fabriziosalmi/proxmox-lxc-autoscale/blob/main/docs/lxc_autoscale/README.md#configuration)

**Comment by pear235** [comment]
Still friday here, i will check it before monday :) Thanks!

**Comment by Bastika07** [comment]
[autoscale_log.txt](https://github.com/user-attachments/files/18937743/autoscale_log.txt)
[lxc_autoscale.yaml.log](https://github.com/user-attachments/files/18937744/lxc_autoscale.yaml.log)


The autoscale is not registering the tier settings.
